### PR TITLE
REALMC-8292 rename to and from flags

### PR DESCRIPTION
**Pull request by kpatel71716**

### CODE REVIEW

File: internal/commands/push/command.go [hunk 1]
```diff
@@ -48,26 +48,26 @@ func (cmd *Command) Inputs() cli.InputResolver {
...
 // Handler is the command handler
 func (cmd *Command) Handler(profile *cli.Profile, ui terminal.UI, clients cli.Clients) error {
-	app, err := local.LoadApp(cmd.inputs.AppDirectory)
+	app, err := local.LoadApp(cmd.inputs.LocalPath)
 	if err != nil {
 		return err
 	}
 
-	to, err := cmd.inputs.resolveTo(ui, clients.Realm)
+	remote, err := cmd.inputs.resolveRemoteApp(ui, clients.Realm)
```
Comment (makesitgo) [inline internal/commands/push/command.go:56]: [nit] use `appRemote` here

File: internal/commands/push/command.go [hunk 2]
```diff
@@ -407,10 +407,10 @@ func deployDraftAndWait(ui terminal.UI, realmClient realm.Client, to to, draftID
 		for deployment.Status == realm.DeploymentStatusCreated || deployment.Status == realm.DeploymentStatusPending {
 			time.Sleep(time.Second)
 
-			deployment, err = realmClient.Deployment(to.GroupID, to.AppID, deployment.ID)
+			deployment, err = realmClient.Deployment(remote.GroupID, remote.AppID, deployment.ID)
 			if err != nil {
-				if e := realmClient.DiscardDraft(to.GroupID, to.AppID, draftID); e != nil {
-					ui.Print(terminal.NewWarningLog("Failed to discard the draft created for your deployment"))
+				if e := realmClient.DiscardDraft(remote.GroupID, remote.AppID, draftID); e != nil {
+					ui.Print(terminal.NewWarningLog("Failed local discard the draft created for your deployment"))
```
Comment (makesitgo) [inline internal/commands/push/command.go:413]: find/replace shenanigans...

File: internal/commands/push/inputs.go [hunk 3]
```diff
@@ -9,16 +9,14 @@ import (
 )
 
 const (
-	flagAppDirectory      = "app-dir"
-	flagAppDirectoryShort = "p"
-	flagAppDirectoryUsage = "provide the path to a Realm app containing the changes to push"
+	flagLocalPath      = "local"
+	flagLocalPathUsage = "provide the path to a Realm app containing the changes to push"
```
Comment (makesitgo) [inline internal/commands/push/inputs.go:13]: "provide the local path..."

File: internal/commands/push/inputs.go [hunk 4]
```diff
@@ -9,16 +9,14 @@ import (
...
 
 	flagProject      = "project"
 	flagProjectUsage = "the MongoDB cloud project id"
 
-	flagTo      = "to"
-	flagToShort = "t"
-	flagToUsage = "choose a Realm app to push changes towards"
+	flagRemote      = "remote"
+	flagRemoteUsage = "choose a Realm app to push changes towards"
```
Comment (makesitgo) [inline internal/commands/push/inputs.go:19]: "specify a remote Realm app..."

**Review by makesitgo — COMMENTED** [review]
first couple of thoughts here...

it won't let me comment on the file, but what were the changes to `internal/commands/app/testdata/project.zip`?

also, after seeing these changes, i'd like to keep the flag literals the same (e.g. `--remote` and `--local`), but lets rename those input fields to be `RemoteApp` and `LocalPath` respectively

(the rest of the names you have, `appRemote`, `pathLocal`, can all remain the same.  this is just for on the `inputs` structs themselves

File: internal/commands/push/command.go (hunk 2 above)
Comment (kpatel71716) [reply internal/commands/push/command.go]: Nice catch! always need a second pair of eyes on this stuff lol

File: internal/commands/app/create.go [hunk 5]
```diff
@@ -27,7 +27,7 @@ type CommandCreate struct {
 func (cmd *CommandCreate) Flags(fs *pflag.FlagSet) {
 	fs.StringVar(&cmd.inputs.Project, flagProject, "", flagProjectUsage)
 	fs.StringVarP(&cmd.inputs.Name, flagName, flagNameShort, "", flagNameUsage)
-	fs.StringVarP(&cmd.inputs.From, flagFrom, flagFromShort, "", flagFromUsage)
+	fs.StringVar(&cmd.inputs.Remote, flagRemote, "", flagRemoteUsage)
 	fs.StringVarP(&cmd.inputs.Directory, flagDirectory, flagDirectoryShort, "", flagDirectoryUsage)
```
[the remark this answers is not in the document]
Comment (kpatel71716) [reply internal/commands/app/create.go]: Tested this out in a separate commit: https://github.com/10gen/realm-cli/pull/91/commits/7ddcecf2967c554334a7060d9415d5e2a3f09994

File: internal/commands/app/create.go [hunk 6]
```diff
@@ -27,8 +27,8 @@ type CommandCreate struct {
 func (cmd *CommandCreate) Flags(fs *pflag.FlagSet) {
 	fs.StringVar(&cmd.inputs.Project, flagProject, "", flagProjectUsage)
 	fs.StringVarP(&cmd.inputs.Name, flagName, flagNameShort, "", flagNameUsage)
-	fs.StringVarP(&cmd.inputs.From, flagFrom, flagFromShort, "", flagFromUsage)
-	fs.StringVarP(&cmd.inputs.Directory, flagDirectory, flagDirectoryShort, "", flagDirectoryUsage)
+	fs.StringVar(&cmd.inputs.RemoteApp, flagRemote, "", flagRemoteUsage)
+	fs.StringVarP(&cmd.inputs.LocalPath, flagLocalPath, flagLocalPathShort, "", flagLocalPathUsage)
```
Comment (makesitgo) [inline internal/commands/app/create.go:31]: i think for consistency's sake (and we may revisit this), let's keep this `--local` with no shorthand (the `inputs` field can still be named `LocalPath`)

File: internal/commands/app/create_inputs.go [hunk 7]
```diff
@@ -16,9 +16,9 @@ import (
 )
 
 var (
-	flagDirectory      = "app-dir"
-	flagDirectoryShort = "p"
-	flagDirectoryUsage = "the directory to create your new Realm app, defaults to Realm app name"
+	flagLocalPath      = "local-path"
+	flagLocalPathShort = "p"
```
Comment (makesitgo) [inline internal/commands/app/create_inputs.go:20]: deleteme

File: internal/commands/app/create_test.go [hunk 8]
```diff
@@ -565,18 +565,18 @@ func TestAppCreateCommandDisplay(t *testing.T) {
...
-				Directory: "realm-app",
+				LocalPath: "realm-app",
 				Cluster:   "Cluster0",
 				DataLake:  "DataLake0",
 				DryRun:    true,
 			},
 		}
 		assert.Equal(t,
-			cli.Name+" app create --project 123 --name test-app --from from-app --app-dir realm-app --location IE --deployment-model LOCAL --cluster Cluster0 --data-lake DataLake0 --dry-run",
+			cli.Name+" app create --project 123 --name test-app --remote remote-app --app-dir realm-app --location IE --deployment-model LOCAL --cluster Cluster0 --data-lake DataLake0 --dry-run",
```
Comment (makesitgo) [inline internal/commands/app/create_test.go:579]: this should probably be failing?  need to update `--app-dir` to `--local` (or change something else to make it fail first ;) )

File: internal/commands/pull/inputs.go [hunk 9]
```diff
@@ -12,19 +12,17 @@ import (
...
 	flagProjectUsage = "the MongoDB cloud project id"
 
 	flagAppVersion      = "app-version"
 	flagAppVersionUsage = "specify the app config version to pull changes down as"
 
-	flagTo      = "to"
-	flagToShort = "t"
-	flagToUsage = "provide the path to export a Realm app to"
+	flagLocal      = "local"
+	flagLocalUsage = "provide the local path to export a Realm app to"
```
Comment (makesitgo) [inline internal/commands/pull/inputs.go:25]: `provide` => `specify`

File: internal/commands/push/inputs.go [hunk 10]
```diff
@@ -9,16 +9,14 @@ import (
 )
 
 const (
-	flagAppDirectory      = "app-dir"
-	flagAppDirectoryShort = "p"
-	flagAppDirectoryUsage = "provide the path to a Realm app containing the changes to push"
+	flagLocalPath      = "local"
+	flagLocalPathUsage = "provide the local path to a Realm app containing the changes to push"
```
Comment (makesitgo) [inline internal/commands/push/inputs.go:13]: [nit] "specify" instead of provide

**Review by makesitgo — APPROVED** [review]
looks good to me once we consolidate everything down to `--local`/`--remote`

thanks for this!

:cool: beans

**Pull request merged**